### DISCUSSION TITLE
Apply CSS Classes to Article element?

### DISCUSSION THREAD

**Discussion by askides** [markdoc/markdoc#99] [Q&A] [answered]
Hello guys!

Is there a way to apply some CSS classes to the `article` element generated by Markdoc?

For example, if you are using Tailwind you need to apply `prose max-w-prose`, but I didn't find a way in the docs.

Any idea?

Thanks :)

**Comment by mfix-stripe** [comment, marked as answer]
@itsrennyman you can create a custom `document` [node](https://markdoc.io/docs/nodes) and define a `class` attribute there. 

Here is an example of a custom document node, which renders an `article`: https://github.com/markdoc/docs/blob/main/markdoc/nodes/document.markdoc.js

**Reply by askides** [reply]
Thank you so much! :)

**Reply by nick-kang** [reply]
We were looking to do the same thing. Super helpful! 

What is `source` and `config.source` in the example? It's throwing a type error in Typescript. Can't seem to find anything about it in the docs.

```typescript
    return new Tag(
      this.render,
      { source: config.source },
      node.transformChildren(config)
    );
```

https://github.com/markdoc/docs/blob/327567184763e9f3c1dfda58e8b89be5978f717d/markdoc/nodes/document.markdoc.js#L13-L17

edit: Looks like it's the component's props (https://github.com/markdoc/docs/blob/main/components/Document.js)

**Reply by mfix-stripe** [reply]
Yup @nick-kang! For `@markdoc/next.js`, we add the plaintext source to the config so that you can access it in your registered Markdoc tags. 

The TypeScript types for it can be found here: https://github.com/markdoc/next.js/blob/main/src/index.d.ts#L14
